packages/@ama-sdk/core/src/fwk/param-deserialization.ts

```diff
@@ -0,0 +1,212 @@
+ */
+function deserializeObjectQueryParams(serializedParamValue: string, paramSerialization: ParamSerialization) {
+  // NOTE: Applies to the exploded styles 'form' and 'deepObject'
+  if (paramSerialization.explode && (paramSerialization.style === 'form' || paramSerialization.style === 'deepObject')) {
+    return serializedParamValue.split('&').reduce((obj: { [key: string]: PrimitiveType }, serializedProperty) => {
+      const [key, value] = serializedProperty.split('=');
+      // NOTE: The key of an object in deepObject style is surrounded by opening and closing square brackets
+      const objKey = paramSerialization.style === 'deepObject' ? key.split(OPENING_SQUARE_BRACKET_URL_CODE)[1].split(CLOSING_SQUARE_BRACKET_URL_CODE)[0] : key;
+      obj[decodeURIComponent(objKey)] = decodeURIComponent(value);
+      return obj;
+    }, {} as { [key: string]: PrimitiveType });
+  }
+
+  // NOTE: Applies to the non-exploded styles 'form', 'spaceDelimited', and 'pipeDelimited'
+  if (paramSerialization.style !== 'deepObject' && !paramSerialization.explode) {
+    // NOTE: The splitParamElements function is called since object query parameters can be split by delimiters based on the serialization style
+    // NOTE: The deserialized value will exist since these exploded styles are supported
+    const splitObject = splitParamElements(serializedParamValue, paramSerialization);
+    return objectFromPairwiseArray(splitObject!);
+  }
+}
+
+/**
+ * Deserialize query parameters based on the values of exploded and style and the parameter type
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedQueryParams serialized query parameters
+ * @param queryParamSerialization query parameter serialization
+ */
+export function deserializeQueryParams<T extends { [key: string]: string }>(
+  serializedQueryParams: T,
+  queryParamSerialization: { [p in keyof T]: ParamSerialization & { paramType: ParamTypeForDeserialization } }
+): { [p in keyof T]: SupportedParamType } {
+  return Object.entries(serializedQueryParams).reduce((acc, [queryParamName, serializedParamValue]) => {
+    const paramSerialization = queryParamSerialization[queryParamName];
+    let deserializedValue: SupportedParamType;
+    if (paramSerialization.paramType === 'array') {
+      deserializedValue = deserializeArrayQueryParams(serializedParamValue, paramSerialization);
+    } else if (paramSerialization.paramType === 'object') {
+      deserializedValue = deserializeObjectQueryParams(serializedParamValue, paramSerialization);
+    } else {
+      // NOTE: Query parameters of type primitive are prefixed with the parameter name like this: 'paramName=value'
+      deserializedValue = decodeURIComponent(serializedParamValue.split('=')[1]);
+    }
+    if (deserializedValue) {
+      acc[queryParamName as keyof T] = deserializedValue;
+    }
+    return acc;
+  }, {} as { [p in keyof T]: SupportedParamType });
```

Comment:
the type is a bit incorrect as this stage (accept if `T` is `{}`).
Maybe it would be more something like:
```suggestion
  }, {} as Record<string, SupportedParamType>) as { [p in keyof T]: SupportedParamType };
```

Reply:
Isn't it fine to accept `{}` ? Because we have conditions in the functions to avoid unsupported use cases
If I change the type to `Record<string, SupportedParamType>`, we could not return an `{}`

Reply:
`{ [p in keyof T]: SupportedParamType }` means that all fields of the object `T` should be present in your object. So if `T` has more than 1 field, the code `{} as { [p in keyof T]: SupportedParamType }` is wrong.

The purpose is not to change the return type (which remains `{ [p in keyof T]: SupportedParamType }`) but the `acc` type to avoid wrong typing.

Reply:
As discussed by message, we decided to use `Partial<...>` objects in order to correctly type the `acc` in the reducers.
This update was applied in both the deserialization and serialization functions.

Reply:
As mentioned in another comment below, this previous message can be ignored.